torch/distributed/distributed_c10d.py

```diff
@@ -1156,6 +1157,148 @@ def all_gather_multigpu(output_tensor_lists,
         work.wait()
 
 
+def _object_to_tensor(obj):
```

Comment:
I didn't see a need to add these the protocols since we have control over most of the serialization here, i.e. we directly serialize and deserialize ourselves. Looking at the [docs](https://docs.python.org/3/library/pickle.html) it seems that specifying a higher protocol might give some performance win over a lower protocol, but I would advocate for keeping this interface simpler if we don't see demand for it.